SharpTox/Core/ToxId.cs

```diff
@@ -64,6 +64,12 @@ public ToxId(byte[] id)
 
         public static bool operator ==(ToxId id1, ToxId id2)
         {
+            if (id1 == null)
```

Comment:
Clearly this will result in a stack overflow.
Also, overloaded operator == implementations should not throw exceptions at all, please fix this.